asapdiscovery-data/asapdiscovery/data/schema/ligand.py

```diff
@@ -229,22 +250,31 @@ def to_oemol(self) -> oechem.OEMol:
                         data[key] = str(getattr(self, key))
         # dump the enum using value to get the str repr
         data["data_format"] = self.data_format.value
-        # dump tags as separate items
-        if self.tags is not None:
-            data.update({k: v for k, v in self.tags.items()})
-        mol = _set_SD_data_repr(mol, data)
+
+        # update conf tags to data
+        data.update(self.conf_tags)
+
+        mol = set_SD_data(mol, data)
+
```

Comment:
Should we not be setting the molecule level tags as well, it looks like they are missing which I think was causing the last testing errors?

Reply:
I think not, if we follow the OpenEye convention as per: https://docs.eyesopen.com/toolkits/python/oechemtk/oemol.html#dude-where-s-my-sd-data

> Practically, it is best to never attach SD tag data to an [OEMCMolBase](https://docs.eyesopen.com/toolkits/python/oechemtk/OEChemClasses/OEMCMolBase.html#OEChem::OEMCMolBase). This should only be done as a space optimization when it is assured that the multi-conformer molecule will only be written to OEB.
> 
> To this end, when an [OEMol](https://docs.eyesopen.com/toolkits/python/oechemtk/OEChemClasses/OEMol.html#oechem-oemol) copy constructs from an [OEGraphMol](https://docs.eyesopen.com/toolkits/python/oechemtk/OEChemClasses/OEGraphMol.html#oechem-oegraphmol) the SD tag data is attached to the first conformer.

Reply:
I think my wording confused things! I mean how will any of the data in `Ligand.tags` be passed to the oemol? If you run the following on this branch you find that only conformer tags are passed on not `Ligand.tags`.
```python
from asapdiscovery.data.backend.openeye import get_SD_data
from asapdiscovery.data.schema.ligand import Ligand

mol = Ligand.from_smiles("CC", compound_name="methane")
mol.tags # should be empty
mol.tags["data"] = 1
mol.conf_tags["help"] = 1
oemol = mol.to_oemol()
get_SD_data(oemol)
```
> {'compound_name': ['methane'],
 'provenance': ['{"isomeric_smiles": "CC", "inchi": "InChI=1S/C2H6/c1-2/h1-2H3", "inchi_key": "OTMSDBZUPAUEDD-UHFFFAOYSA-N", "fixed_inchi": "InChI=1/C2H6/c1-2/h1-2H3", "fixed_inchikey": "OTMSDBZUPAUEDD-UHFFFAOYNA-N"}'],
 'data_format': ['sdf'],
 'help': ['1']}

Reply:
Ah I see what you mean! My suggestion would be to avoid this issue by using `set_SD_data` instead:
```
from asapdiscovery.data.backend.openeye import get_SD_data
from asapdiscovery.data.schema.ligand import Ligand

mol = Ligand.from_smiles("CC", compound_name="methane")
mol.tags # should be empty
mol.set_SD_data({'mydata': 1})
oemol = mol.to_oemol()
get_SD_data(oemol)
```

> {'compound_name': ['methane'],
 'provenance': ['{"isomeric_smiles": "CC", "inchi": "InChI=1S/C2H6/c1-2/h1-2H3", "inchi_key": "OTMSDBZUPAUEDD-UHFFFAOYSA-N", "fixed_inchi": "InChI=1/C2H6/c1-2/h1-2H3", "fixed_inchikey": "OTMSDBZUPAUEDD-UHFFFAOYNA-N"}'],
 'data_format': ['sdf'],
 'help': ['1'],
 'mydata': ['1']}

Reply:
if you use `set_SD_data`, it will also warn you that:
`UserWarning: Tag name data is a reserved attribute name, skipping`

Reply:
okay that was a bad example using data as a tag. What about tags passed in the init like this 
```python
from asapdiscovery.data.backend.openeye import get_SD_data
from asapdiscovery.data.schema.ligand import Ligand

mol = Ligand.from_smiles("CC", compound_name="methane", random_data="save_me")
oemol = mol.to_oemol()
get_SD_data(oemol)
```
> {'compound_name': ['methane'],
 'provenance': ['{"isomeric_smiles": "CC", "inchi": "InChI=1S/C2H6/c1-2/h1-2H3", "inchi_key": "OTMSDBZUPAUEDD-UHFFFAOYSA-N", "fixed_inchi": "InChI=1/C2H6/c1-2/h1-2H3", "fixed_inchikey": "OTMSDBZUPAUEDD-UHFFFAOYNA-N"}'],
 'data_format': ['sdf']}

Reply:
Yeah ok obviously you're right, I've added 1 line that makes that happen. We'll see if that breaks everything else:
```
from asapdiscovery.data.schema.ligand import Ligand
mol = Ligand.from_smiles("CC", compound_name="methane", random_data="save_me")
oemol = mol.to_oemol()
get_SD_data(oemol)
Out[2]: 
{'compound_name': ['methane'],
 'provenance': ['{"isomeric_smiles": "CC", "inchi": "InChI=1S/C2H6/c1-2/h1-2H3", "inchi_key": "OTMSDBZUPAUEDD-UHFFFAOYSA-N", "fixed_inchi": "InChI=1/C2H6/c1-2/h1-2H3", "fixed_inchikey": "OTMSDBZUPAUEDD-UHFFFAOYNA-N"}'],
 'data_format': ['sdf'],
 'random_data': ['save_me']}
```